lntest/itest/lnd_forward_interceptor_test.go

```diff
@@ -29,166 +31,136 @@ type interceptorTestCase struct {
-// 		tc.t.Fatalf("failed to disconnect alice and bob")
-// 	}
-// 	tc.net.EnsureConnected(tc.t.t, tc.bob, tc.alice)
-
-// 	// Here we wait for the channel to be active again.
-// 	err = wait.Predicate(func() bool {
-// 		req := &lnrpc.ListChannelsRequest{
-// 			ActiveOnly: true,
-// 			Peer:       tc.alice.PubKey[:],
-// 		}
-
-// 		channels, err := tc.bob.ListChannels(ctxt, req)
-// 		return err == nil && len(channels.Channels) > 0
-// 	}, defaultTimeout)
-// 	require.NoError(
-// 		tc.t.t, err, "alice <> bob channel didn't re-activate",
-// 	)
-
-// 	// Now that the channel is active we make sure the test passes as
-// 	// expected.
-// 	payments, err := tc.alice.ListPayments(ctxb, &lnrpc.ListPaymentsRequest{
-// 		IncludeIncomplete: true,
-// 	})
-// 	require.NoError(tc.t.t, err, "failed to fetch payment")
-
-// 	// We expect one in flight payment since we held the htlcs.
-// 	require.Equal(tc.t.t, len(payments.Payments), 1)
-// 	require.Equal(tc.t.t, payments.Payments[0].Status, lnrpc.Payment_IN_FLIGHT)
-
-// 	// We now fail all htlcs to cancel the payment.
-// 	packetsCount := 0
-// 	interceptedPacketstMap.Range(func(_, packet interface{}) bool {
-// 		p := packet.(*routerrpc.ForwardHtlcInterceptRequest)
-// 		_ = interceptor.Send(&routerrpc.ForwardHtlcInterceptResponse{
-// 			IncomingCircuitKey: p.IncomingCircuitKey,
-// 			Action:             routerrpc.ResolveHoldForwardAction_FAIL,
-// 		})
-// 		packetsCount++
-// 		return true
-// 	})
-
-// 	// At this point if we have more than one held htlcs then we should
-// 	// fail.  This means we hold the same htlc twice which is a risk we
-// 	// want to eliminate. If we don't have the same htlc twice in theory we
-// 	// can cancel one and settle the other by mistake.
-// 	require.Equal(tc.t.t, packetsCount, 1)
-
-// 	cancelInterceptor()
-// 	wg.Wait()
-// }
+// testForwardInterceptorDedupHtlc tests that upon reconnection, duplicate
+// HTLCs aren't re-notified using the HTLC interceptor API.
+func testForwardInterceptorDedupHtlc(ht *lntest.HarnessTest) {
+	// Initialize the test context with 3 connected nodes.
+	ctx := newInterceptorTestContext(ht)
+	defer ctx.shutdownNodes()
+
+	const chanAmt = btcutil.Amount(300000)
+
+	// Open and wait for channels.
+	chanPoint := ctx.openChannel(ctx.alice, ctx.bob, chanAmt)
+	ctx.openChannel(ctx.bob, ctx.carol, chanAmt)
+	defer ctx.closeChannels()
+
+	// Connect the interceptor.
+	interceptor, cancelInterceptor := ht.HtlcInterceptor(ctx.bob)
+
+	// Prepare the test cases.
+	req := &lnrpc.Invoice{ValueMsat: 1000}
+	addResponse := ht.AddInvoice(req, ctx.carol)
+	invoice := ht.LookupInvoice(ctx.carol, addResponse.RHash)
+	tc := &interceptorTestCase{
+		amountMsat: 1000,
+		invoice:    invoice,
+		payAddr:    invoice.PaymentAddr,
+	}
+
+	// We initiate a payment from Alice.
+	done := make(chan struct{})
+	go func() {
+		ctx.sendPaymentAndAssertAction(tc)
+
+		// Signal that all the payments have been sent.
+		close(done)
```

Comment:
This should be a `defer close(done)` since any `require` failure inside `sendPaymentAndAssertAction` will exit the goroutine with only calling `defer` statements.

Reply:
Good catch, now fixed!